### PR TITLE
Add bindings for git_revert{,_commit}

### LINKED ISSUE #571: add support for `git revert`
Support for `git revert` seems not to be implemented yet. I couldn't find anything relevant in the docs. Also, grepping the source for libgit2's `git_revert` didn't yield any results.

I would find it useful to add support for `git revert`. Please consider adding it.

### PR DESCRIPTION
**Pull request by toxeus**
Closes #571

### CODE REVIEW

File: src/repo.rs [hunk 1]
```diff
@@ -2721,6 +2722,46 @@ impl Repository {
             Ok(())
         }
     }
+
+    /// Perform a revert.
```
Comment (extrawurst) [inline src/repo.rs:2726]: looking at the two comments it sounds misleading as If this is not changing the index. reading up on https://libgit2.org/libgit2/#HEAD/group/revert/git_revert and https://libgit2.org/libgit2/#HEAD/group/revert/git_revert_commit it sounds more like both will alter the index. 
some of the methods in git2-rs went about plain duplicating the doc comment from the original. maybe this makes sense here aswell?

Comment (toxeus) [reply src/repo.rs:2726]: Yes, both change the index. One changes only the index. Will duplicate the doc comment :+1:

**Comment by extrawurst** [comment]
@alexcrichton I think this looks good to go

**Comment by alexcrichton** [comment]
Thanks!

**Pull request merged**